src/credentials/examples/DeviceAttestationVerifierExample.cpp

```diff
@@ -0,0 +1,221 @@
+}
+
+class ExampleDACVerifier : public DeviceAttestationVerifier
+{
+public:
+    AttestationVerificationResult
+    VerifyAttestationInformation(const ByteSpan & attestation_info_buffer, const ByteSpan & attestation_challenge_buffer,
+                                 const ByteSpan & attestation_signature_buffer, const ByteSpan & pai_cert_der_buffer,
+                                 const ByteSpan & dac_cert_der_buffer, const ByteSpan & attestation_nonce) override;
+};
+
+AttestationVerificationResult ExampleDACVerifier::VerifyAttestationInformation(const ByteSpan & attestation_info_buffer,
+                                                                               const ByteSpan & attestation_challenge_buffer,
+                                                                               const ByteSpan & attestation_signature_buffer,
+                                                                               const ByteSpan & pai_cert_der_buffer,
+                                                                               const ByteSpan & dac_cert_der_buffer,
+                                                                               const ByteSpan & attestation_nonce)
+{
+    // match DAC and PAI VIDs
+    if (!pai_cert_der_buffer.empty())
+    {
+        VendorId paiVid;
+        VendorId dacVid;
+
+        CHIP_ERROR error = ExtractVIDFromX509Cert(pai_cert_der_buffer, paiVid);
+        VerifyOrReturnError(error == CHIP_NO_ERROR || error == CHIP_ERROR_KEY_NOT_FOUND,
+                            AttestationVerificationResult::kPaiFormatInvalid);
+
+        if (error == CHIP_NO_ERROR)
+        {
+            VerifyOrReturnError(ExtractVIDFromX509Cert(dac_cert_der_buffer, dacVid) == CHIP_NO_ERROR,
+                                AttestationVerificationResult::kDacFormatInvalid);
+
+            VerifyOrReturnError(paiVid == dacVid, AttestationVerificationResult::kDacVendorIdMismatch);
+        }
+    }
+
+    P256PublicKey remoteManufacturerPubkey;
+    VerifyOrReturnError(ExtractPubkeyFromX509Cert(dac_cert_der_buffer, remoteManufacturerPubkey) == CHIP_NO_ERROR,
+                        AttestationVerificationResult::kDacFormatInvalid);
+
+    // Step d
+    P256ECDSASignature deviceSignature;
+    // SetLength will fail if signature doesn't fit
+    VerifyOrReturnError(deviceSignature.SetLength(attestation_signature_buffer.size()) == CHIP_NO_ERROR,
+                        AttestationVerificationResult::kInvalidSignature);
+    memcpy(deviceSignature.Bytes(), attestation_signature_buffer.data(), attestation_signature_buffer.size());
+    VerifyOrReturnError(ValidateAttestationSignature(remoteManufacturerPubkey, attestation_info_buffer,
+                                                     attestation_challenge_buffer, deviceSignature) == CHIP_NO_ERROR,
+                        AttestationVerificationResult::kDacSignatureInvalid);
```

Comment:
This should be `kAttestationSignatureInvalid`. `kDacSignatureInvalid` sounds to me like "the signature on the DAC certificate itself was invalid based on the payload of the DAC".